controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -0,0 +1,111 @@
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=toolchainclusters,verbs=get;list;watch
+
+// Reconcile ensures that SpaceProvisionerConfig is valid and points to an existing ToolchainCluster.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	spaceProvisionerConfig := &toolchainv1alpha1.SpaceProvisionerConfig{}
+	if err := r.Client.Get(ctx, request.NamespacedName, spaceProvisionerConfig); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("SpaceProvisionerConfig not found anymore")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get SpaceProvisionerConfig: %w", err)
+	}
+
+	if util.IsBeingDeleted(spaceProvisionerConfig) {
+		logger.Info("SpaceProvisionerConfig is being deleted - skipping reconciliation")
+		return ctrl.Result{}, nil
+	}
+
+	// check that there is a ToolchainCluster CR in the same namespace
+	toolchainCluster := &toolchainv1alpha1.ToolchainCluster{}
+	toolchainClusterKey := runtimeclient.ObjectKey{Name: spaceProvisionerConfig.Spec.ToolchainCluster, Namespace: spaceProvisionerConfig.Namespace}
+	toolchainPresent := corev1.ConditionTrue
+	toolchainPresenceReason := toolchainv1alpha1.SpaceProvisionerConfigValidReason
+	var reportedError error
+	toolchainPresenceMessage := ""
+	if err := r.Client.Get(ctx, toolchainClusterKey, toolchainCluster); err != nil {
+		if !errors.IsNotFound(err) {
+			// we need to requeue the reconciliation in this case because we cannot be sure whether the ToolchainCluster
+			// is really present in the cluster or not. If we did not do that and instead just reported the error in
+			// the status, we could eventually leave the SPC in an incorrect state once the error condition in the cluster,
+			// that prevents us from reading the ToolchainCluster, clears. I.e. we need the requeue to keep the promise
+			// of eventual consistency.
+
+			reportedError = fmt.Errorf("failed to get the referenced ToolchainCluster: %w", err)
+			toolchainPresenceMessage = reportedError.Error()
+		}
+		toolchainPresenceReason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason
+		toolchainPresent = corev1.ConditionFalse
+	}
+
+	spaceProvisionerConfig.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  toolchainPresent,
+			Reason:  toolchainPresenceReason,
+			Message: toolchainPresenceMessage,
+		})
+
+	if err := r.Client.Status().Update(ctx, spaceProvisionerConfig); err != nil {
+		if reportedError != nil {
+			logger.Info("failed to update the status (reported as failed reconciliation) with a previous unreported error during reconciliation", "unreportedError", reportedError)
+		}
+		reportedError = fmt.Errorf("failed to update the SpaceProvisionerConfig status: %w", err)
```

Comment:
Thanks for changing this.
In most of the other controllers, we have some helper functions that help us with dealing with status updates and the error hanging
eg: 
https://github.com/codeready-toolchain/host-operator/blob/06236ebe82a9d4150226ac179b1f4bbdc1a0324f/controllers/space/space_controller.go#L672-L686

But it's totally fine as you did - it's a simple controller. If it would get more complicated, then we could extract the logic.

TBH, writing more generic helper functions for the status updates that could be used for all toolchain CRDs has been my plan for a long time already.